File: exercise/echo-function/evaluation/Evaluator.kt
class Evaluator {
    companion object {
        @JvmStatic
        fun evaluate(actual: Any?): EvaluationResult {
            return EvaluationResult.Builder(result = "correct" == actual,
                    readableExpected = actual.toString(),
                    readableActual = actual?.toString() ?: "")
                    .withMessage(EvaluationResult.Message("Hallo"))
                    .build()
        }

        @JvmStatic
        fun evaluateValue(expected: Any, actual: Any?, arguments: List<Any?>?): EvaluationResult {
            return EvaluationResult.Builder(result = expected == actual,
                    readableExpected = expected.toString(),
                    readableActual = actual?.toString() ?: "")
                    .withMessage(EvaluationResult.Message("Hallo"))
                    .build()
        }
    }
}

File: exercise/echo-function/evaluation/Evaluator.cs
using Tested;
using System.Collections;

public class Evaluator {
    public static EvaluationResult Evaluate(Object actual) {
        var correct = "correct" == actual;
        var messages = new List<Message>() {new Tested.Message("Hallo")};
        return new EvaluationResult(correct, "correct", actual != null ? actual.ToString() : "", messages);
    }

    public static EvaluationResult EvaluateValue(Object expected, Object actual, IList arguments) {
        var messages = new List<Message>() {new Tested.Message("Hallo")};
        return new EvaluationResult(expected == actual, expected.ToString(), actual != null ? actual.ToString() : "", messages);
    }
}
